### PR TITLE
Update documentation on Gap Junctions

### LINKED ISSUE #2189: Update documentation on gap junctions
**Describe the flaw**
The documentation for the gap junctions guide specifically refers to NEST 2.x .
https://nest-simulator.readthedocs.io/en/v3.1/guides/simulations_with_gap_junctions.html

**Suggest an improvement**
Ensure that the example works in NEST 3.x and update the text if anything is no longer relevant

### PR DESCRIPTION
**Pull request by jessica-mitchell**
The code used in 'simulations with gap junctions' guide has not been updated to be functional with NEST > 3.

To make the code work, I made changes based on what was used in the inhibitory gap junction example (examples for gap junction do work for NEST 3).
References and minor format issues fixed up as well.

Fixes #2189

### CODE REVIEW

File: doc/htmldoc/synapses/simulations_with_gap_junctions.rst [hunk 1]
```diff
@@ -92,18 +90,16 @@ total number of gap junctions should be chosen as
...
 
-For simulations with gap junctions NEST uses an iterative solution
+For simulations with gap junctions, NEST uses an iterative solution
 scheme based on a numerical method called Jacobi waveform relaxation.
 The default settings of the iterative method are based on numerical
-results, benchmarks and previous experience with gap-junction
-simulations (see `Hahne et al.
-2015 <http://journal.frontiersin.org/article/10.3389/fninf.2015.00022/full>`__)
+results, benchmarks, and previous experience with gap-junction
+simulations (see Hahne et al. [2]_)
```
Comment (clinssen) [inline doc/htmldoc/synapses/simulations_with_gap_junctions.rst:102]:
```suggestion
simulations [2]_.
```

File: doc/htmldoc/synapses/simulations_with_gap_junctions.rst [hunk 2]
```diff
@@ -61,25 +55,29 @@ level with e.g. the ``random`` module of the Python Standard Library:
     # total number of gap junctions
     n_gap_junction = 3000
 
+    gap_weight = 0.5
     n = nest.Create('hh_psc_alpha_gap', n_neuron)
```
Comment (clinssen) [inline doc/htmldoc/synapses/simulations_with_gap_junctions.rst:59]: Mix of single and double quotation marks.

File: doc/htmldoc/synapses/simulations_with_gap_junctions.rst [hunk 3]
```diff
@@ -29,9 +21,11 @@ possibility to create both connections with a single call to
 
     a = nest.Create('hh_psc_alpha_gap')
     b = nest.Create('hh_psc_alpha_gap')
+    gap_weight = 0.5
+    syn_dict = {'synapse_model':'gap_junction', 'weight': gap_weight}
```
Comment (clinssen) [inline doc/htmldoc/synapses/simulations_with_gap_junctions.rst:25]:
```suggestion
    syn_dict = {'synapse_model': 'gap_junction', 'weight': gap_weight}
```

**Review by clinssen — CHANGES_REQUESTED** [review]

File: doc/htmldoc/synapses/simulations_with_gap_junctions.rst [hunk 4]
```diff
@@ -27,17 +19,19 @@ possibility to create both connections with a single call to
 
     import nest
 
-    a = nest.Create('hh_psc_alpha_gap')
-    b = nest.Create('hh_psc_alpha_gap')
+    a = nest.Create("hh_psc_alpha_gap")
+    b = nest.Create("hh_psc_alpha_gap")
+    gap_weight = 0.5
+    syn_dict = {"synapse_model":"gap_junction", "weight": gap_weight}
```
Comment (jessica-mitchell) [inline doc/htmldoc/synapses/simulations_with_gap_junctions.rst:25]:
```suggestion
    syn_dict = {"synapse_model": "gap_junction", "weight": gap_weight}
```

File: doc/htmldoc/synapses/simulations_with_gap_junctions.rst (hunk 2 above)
Comment (jessica-mitchell) [reply doc/htmldoc/synapses/simulations_with_gap_junctions.rst]: all now  "

**Review by clinssen — APPROVED** [review]

**Comment by jessica-mitchell** [comment]
Merging with 1 review as its doc related

**Pull request merged**